src/chainstate/burn/db/sortdb.rs

```diff
@@ -6265,7 +6425,99 @@ pub mod tests {
             .expect("Failed to get peg-in ops from sortition DB");
 
         assert_eq!(res_peg_ins.len(), 1);
-        assert_eq!(res_peg_ins[0].amount, 1337);
+        assert_eq!(res_peg_ins[0], peg_in);
+    }
+
+    #[test]
+    fn test_insert_peg_out_request() {
+        let txid = Txid([0; 32]);
+        let block_height = 123;
+        let vtxindex = 456;
+        let amount = 1337;
+        let recipient = PoxAddress::Addr32(false, address::PoxAddressType32::P2TR, [0; 32]);
+        let signature = MessageSignature([0; 65]);
+        let burn_header_hash = BurnchainHeaderHash([0x03; 32]);
+
+        let peg_out_request = PegOutRequestOp {
+            recipient,
+            amount,
+            signature,
+
+            txid,
+            vtxindex,
+            block_height,
+            burn_header_hash,
+        };
+
+        let first_burn_hash = BurnchainHeaderHash::from_hex(
+            "0000000000000000000000000000000000000000000000000000000000000000",
+        )
+        .unwrap();
+
+        let epochs = StacksEpoch::unit_test(StacksEpochId::Epoch21, block_height);
+        let mut db =
+            SortitionDB::connect_test_with_epochs(block_height, &first_burn_hash, epochs).unwrap();
+
+        let snapshot = test_append_snapshot(
+            &mut db,
+            BurnchainHeaderHash([0x01; 32]),
+            &vec![BlockstackOperationType::PegOutRequest(
+                peg_out_request.clone(),
+            )],
+        );
+
+        let res_peg_out_requests =
+            SortitionDB::get_peg_out_request_ops(db.conn(), &burn_header_hash)
+                .expect("Failed to get peg-in ops from sortition DB");
+
+        assert_eq!(res_peg_out_requests.len(), 1);
+        assert_eq!(res_peg_out_requests[0], peg_out_request);
+    }
```

Comment:
Please expand both of these tests to cover the case where the same transaction gets included in two different burnchain blocks.  The operation metadata, such as the `block_height` and `vtxindex`, should be different in both cases but the operation itself should be the same.  The test should confirm that different metadata are stored for different burnchain blocks.

Reply:
Done :heavy_check_mark: